Fix the openshift templates

metlos opened the pull request:
* Specify that the postgresql images should be taken from openshift
* Use an init container in the rails-postgresql template to init the database

metlos committed:
Fix the openshift templates

* Specify that the postgresql images should be taken from openshift
* Use an init container in the rails-postgresql template to init the
database

phracek commented on openshift/templates/rails-postgresql-persistent.json:
```diff
@@ -408,7 +408,7 @@
           "description": "Defines how to deploy the database",
           "template.alpha.openshift.io/wait-for-ready": "true",
-          "image.openshift.io/triggers": "[{\"from\":{\"kind\":\"ImageStreamTag\",\"name\":\"postgresql:${POSTGRESQL_VERSION}\"},\"fieldPath\": \"spec.template.spec.containers[0].image\"}]"
+          "image.openshift.io/triggers": "[{\"from\":{\"kind\":\"ImageStreamTag\",\"name\":\"postgresql:${POSTGRESQL_VERSION}\", \"namespace\": \"openshift\"},\"fieldPath\": \"spec.template.spec.containers[0].image\"}]"
```
This is not needed. Before testing you are already in required 'namespace'

phracek replied:
Or use variable https://github.com/sclorg/rails-ex/blob/master/openshift/templates/rails-postgresql-persistent.json#L522, that is better approach

phracek commented on openshift/templates/rails-postgresql.json:
```diff
@@ -316,7 +396,7 @@
           "description": "Defines how to deploy the database",
           "template.alpha.openshift.io/wait-for-ready": "true",
-          "image.openshift.io/triggers": "[{\"from\":{\"kind\":\"ImageStreamTag\",\"name\":\"postgresql:12-el8\"},\"fieldPath\": \"spec.template.spec.containers[0].image\"}]"
+          "image.openshift.io/triggers": "[{\"from\":{\"kind\":\"ImageStreamTag\",\"name\":\"postgresql:12-el8\", \"namespace\": \"openshift\"},\"fieldPath\": \"spec.template.spec.containers[0].image\"}]"
```
The same here with 'namespace'

phracek commented:
[test]

metlos committed:
Use the NAMESPACE and POSTGRESQL_VERSION for finding the postgresql image

and remove the recreteParams that were not exactly working well. The
intended function is done by the initContainer.

phracek reviewed (approved):
Thanks for updating this PR. Well Done.
